packages/block-library/src/navigation/interactivity.js

```diff
@@ -51,6 +54,23 @@ store( {
 						? 'dialog'
 						: '';
 				},
+				ariaHiddenAttribute: ( { context } ) => {
+					const { isMenuOpen, globalNav } = context.core.navigation;
+					if ( globalNav ) {
+						let isGlobalNavWrapped = false;
+						// If the Open menu button is visible, set `isGlobalNavWrapped` to true.
+						if (
+							globalNav
+								.querySelector(
+									'button.wp-block-navigation__responsive-container-open '
+								)
+								?.checkVisibility()
```

Comment:
Oh, I think this is [not supported on Safari](https://caniuse.com/mdn-api_element_checkvisibility) yet.

Also, I wonder if this is the best way to check this out. Maybe using `matchMedia` and the value of the `overlayMenu` attribute?

Reply:
If I'm not mistaken, the breakpoint is currently [hardcoded here](https://github.com/WordPress/gutenberg/blob/trunk/packages/base-styles/_breakpoints.scss#L10) to `782px`.

We could use it in `matchMedia`, but if someone modifies it in the future, it may break this. It doesn't seem to be available in PHP, so maybe we should avoid hardcoding that ourselves.

Reply:
> I couldn't find any other way of checking this because, if I am not mistaken, it is handled with CSS.

Oh, I don't think the hamburger menu is handled by CSS, is it? Only the submenus are.


https://github.com/WordPress/gutenberg/assets/3305402/551587c6-ff9f-48c7-9f1f-486bb4f8f005

Reply:
> Oh, I don't think the hamburger menu is handled by CSS, is it? Only the submenus are.

What I meant is that, to identify if we should wrap the menu or not, depending on the screen, is handled by CSS.

* If it is wrapped (screen < 600px) -> `aria-hidden` should be true.
* If it is not wrapped (screen > 600px) -> `aria-hidden` should be false.


https://github.com/WordPress/gutenberg/assets/34552881/cccc9f13-44eb-4ca4-b809-f4bc350dfc68

> We could use it in matchMedia, but if someone modifies it in the future, it may break this. It doesn't seem to be available in PHP, so maybe we should avoid hardcoding that ourselves.

Apart from that, other users/plugins could be modifying that breakpoint somehow right? (Maybe using `!important` I don't know 🤷 ).

> Oh, I think this is [not supported on Safari](https://caniuse.com/mdn-api_element_checkvisibility) yet.

I didn't check that, sorry. You're totally right. I believe we can use the [`offsetParent` property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent). As explained there: _"offsetParent returns null in the following situations: The element or any ancestor has the display property set to none."_

Reply:
I just made a commit using the `offsetParent` property.

On the other hand, I was reviewing it with @DAreRodz and, according to [MDN docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden): _"aria-hidden="true" should not be added when the element or the element's ancestor is hidden with [display: none](https://developer.mozilla.org/en-US/docs/Web/CSS/display)"_.

If I remember correctly, I added it because the current implementation has it. Should we just remove the `aria-hidden` property?

Reply:
I finally made another commit to remove directly the `aria-hidden` attribute because it seems that it shouldn't be used in this use case according to [MDN docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden#:~:text=aria%2Dhidden%3D%22true%22%20should%20not%20be%20added,the%20element%27s%20ancestor%20is%20hidden%20with%20visibility%3A%20hidden).

Reply:
Just to confirm: this is better than what the original implementation is doing (which is adding it the first time the menu is opened), right? When `display: none` is set, `aria-hidden="true"` is unnecessary, and when `display` is different than `none`, `aria-hidden="false"` is irrelevant.

Reply:
Oh, confirmed by @alexstine below. Perfect 🙂